src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -162,15 +164,16 @@ def utc_dataframe_to_aligned(
 @pa.check_types
 def pivot_aligned_timeseries_dataframe(
     aligned_df: DataFrame[AlignedTimeseriesDataFrame],
-    periods: int,
+    periods: int = 24,
+    values_col: str = "value_col",
 ) -> DataFrame[TimeseriesMatrix]:
     """Pivot aligned timeseries dataframe into timeseries matrix and pad if needed.
 
     Padding finds the complete list of hours from the start of the first day
     present in the timeseries to the end of the last, and then fills any missing hours
     with NULLs.
     """
-    matrix = aligned_df.pivot(index="datetime", columns="id_col", values="value_col")
+    matrix = aligned_df.pivot(index="datetime", columns="id_col", values=values_col)
 
     # Pad matrix with any missing hours from timeseries
```

Comment:
If I understand this correctly, it's potentially changing the number of rows in the dataframe -- if it does end up filling in hours at the start or end, or gaps in the middle.  So what happens with the reindex at the end?  Is that what causes NULL values to be added, in association with any new index value that wasn't present in the original?  If so, a comment explaning that would be helpful

Reply:
Yeah the `reindex` is what actually adds the NULL's. I've added a comment explaining.